### ISSUE TITLE
Error while Running Disk Image

### ISSUE THREAD

**Issue by not-nef** [sumeshir26/TimerX#20]
ok, so, i tried building a disk image from the current repo state (i only excluded the line mentioned in the previous issue) and when running the app, it throws me the following error:

```

Process:               TimerX [967]
Path:                  /Volumes/VOLUME/TimerX.app/Contents/MacOS/TimerX
Identifier:            TimerX
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           TimerX [967]
User ID:               501

Date/Time:             2021-12-20 20:14:28.654 +0100
OS Version:            Mac OS X 10.13.6 (17G14042)
Report Version:        12
Anonymous UUID:        5A231B38-67FD-F6BD-0CEC-C01BB4BA3048


Time Awake Since Boot: 190 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x5] Code Signature

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: @executable_path/lib/Python
  Referenced from: /Volumes/VOLUME/TimerX.app/Contents/MacOS/TimerX
  Reason: no suitable image found.  Did find:
	/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python: code signature invalid for '/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python'

	/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python: stat() failed with errno=1
	/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python: code signature invalid for '/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python'

	/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python: stat() failed with errno=1

Binary Images:
       0x10fc18000 -        0x10fc19ffb +TimerX (0) <856E75FB-2265-3365-96A2-A3FB7DAE57BB> /Volumes/VOLUME/TimerX.app/Contents/MacOS/TimerX
       0x1177b1000 -        0x1177fbadf  dyld (551.5) <CB9BFB56-4511-36F1-A546-891FF770C01C> /usr/lib/dyld

Model: Macmini5,1, BootROM 135.0.0.0.0, 2 processors, Intel Core i5, 2,3 GHz, 4 GB, SMC 1.76f0
Graphics: Intel HD Graphics 3000, Intel HD Graphics 3000, Built-In
Memory Module: BANK 0/DIMM0, 2 GB, DDR3, 1067 MHz, 0x830B, 0x4E54324743363442384841314E532D424520
Memory Module: BANK 1/DIMM0, 2 GB, DDR3, 1067 MHz, 0x830B, 0x4E54324743363442384841314E532D424520
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0xE4), Broadcom BCM43xx 1.0 (5.106.98.102.30)
Bluetooth: Version 6.0.7f22, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en1
Serial ATA Device: Samsung SSD 840 EVO 250GB, 250,06 GB
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: Extreme SSD
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: Gaming Keyboard
USB Device: USB Gaming Mouse
USB Device: IR Receiver
Thunderbolt Bus: Mac mini, Apple Inc., 25.1
```

F

**Comment by im-coder-lg** [comment]
So it's a Mac Mini... Can it run Catalina?

**Comment by im-coder-lg** [comment]
I don't understand what this means:
```console
Dyld Error Message:
  Library not loaded: @executable_path/lib/Python
  Referenced from: /Volumes/VOLUME/TimerX.app/Contents/MacOS/TimerX
  Reason: no suitable image found.  Did find:
	/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python: code signature invalid for '/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python'

	/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python: stat() failed with errno=1
	/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python: code signature invalid for '/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python'

	/Volumes/TimerX/TimerX.app/Contents/MacOS/lib/Python: stat() failed with errno=1
```
Is it saying that it couldn't find the Python library?

**Comment by sumeshir26** [comment]
@im-coder-lg Yeah, python doesent seem to run, now running a action which will generate a DMG to give to @not-nef

**Comment by im-coder-lg** [comment]
:+1:

**Comment by sumeshir26** [comment]
[Uploading TimerX(1).zip…]()
@not-nef Try This

**Comment by im-coder-lg** [comment]
Link doesn't work.

**Comment by sumeshir26** [comment]
@not-nef Try this:
[TimerX(1).zip](https://github.com/sumeshir26/TimerX/files/7749380/TimerX.1.zip)

**Comment by not-nef** [comment]
> So it's a Mac Mini... Can it run Catalina?

no, it cant run any newer version then it already does.

**Comment by sumeshir26** [comment]
@not-nef Run that DMG(It should work)

**Comment by im-coder-lg** [comment]
It's just a redirect, try uploading the DMG but this time, use the popup 
method.

**Comment by not-nef** [comment]
> @not-nef Try this:
> 
> [TimerX(1).zip](https://github.com/sumeshir26/TimerX/files/7749380/TimerX.1.zip)

It says i need macos 10.14 or newer.

ill MAYBE be able to test it on a mac with big sur, but i cant promise.

**Comment by im-coder-lg** [comment]
> It says i need macos 10.14 or newer.
Wow. High Sierra ain't supported anymore. @not-nef best is to switch to 
Linux since my switching will take a LOT of time(lazy me).

 > ill MAYBE be able to test it on a mac with big sur, but i cant promise.

Well, if it is kind of like a company laptop, DO NOT try the DMG on it. I 
bet they'll deactivate the laptop after noticing an "unknown" program on 
it.

**Comment by not-nef** [comment]
epic quote fail

no its not a company laptop, its the macmini of my dad.

**Comment by im-coder-lg** [comment]
Does it have Big Sur? If it does, tell your dad that you want to test an installer of an open-source project and that you will uninstall it after the test, since that could relieve your dad's mind about viruses and it will free up some disk space.

**Comment by im-coder-lg** [comment]
Also, noticed this(you forgot to delete your edit):
> epic quote fail

I am not offended, it's Vivaldi Mail. Anyway, back to the topic. Should we add a Python Library in the so-called `@executable-path`? Moreover @sumeshir26 can you run the release draft action of Windows and macOS? I wanna try a teardown with PeaZip. You don't need to send a link to download, you can just run the action and inform me, I'll take care of the rest.

**Comment by sumeshir26** [comment]
In the actions tab, Look at artifacts produced in the latest run of the Release Build workflow.

**Comment by sumeshir26** [comment]
You know what, I will give you write acess to this repo so you can do these things without waiting for me.

**Comment by im-coder-lg** [comment]
I didn't recieve the notif yet, can you check that?

**Comment by not-nef** [comment]
Bad news, i tried running the app on a big sur mac. If i open it, you can see TimerX in the Menu bar (the gui doesnt load tho) and after a split second it just closes.

i suggest dropping support for macos.

**Comment by im-coder-lg** [comment]
We need some logs, can you try making a recording + install logs to help us in this? Also, I will have to use PeaZip for a teardown so prepare for that too!

**Comment by not-nef** [comment]
this is getting a bit too complicated for me, and i wont install anything on my dads mac as he doesnt like that (not because he doesnt know anything about it, but he just doesnt want it dont ask why).

**Comment by not-nef** [comment]
god why is macos so important to you guys

**Comment by im-coder-lg** [comment]
> god why is macos so important to you guys

Well, if you look at the computer(desktop/laptop) share in today's market and just compare it to the air share, Windows is like Nitrogen, macOS is like Oxygen and Linux is like the remaining 1% of mixed gases(different distros). We could use the Wine compatibility layer but that needs to be installed via HomeBrew. So we thought of making MSIs, which are successful, then DMGs, RPMs and using a Linux tool called Alien, convert the RPM to DEB.
Maybe we could just ask users to use the Wine layer till we successfully add support.

**Comment by sumeshir26** [comment]
@not-nef Thanks for your help, will tell when MacOS support is re-added.

**Comment by sumeshir26** [comment]
@im-coder-lg Continue with your teardown...

**Comment by im-coder-lg** [comment]
Hey @sumeshir26 since we have separate actions for each OS, can we remove the old `release.yml` too? Best done in @not-nef's fork since that'd be easier.

**Comment by sumeshir26** [comment]
No! If you run that, it successfully generates one for all platforms and creates a release successfully. I spent a week on that!

**Comment by im-coder-lg** [comment]
Well, didn't do that! Phew!

**Comment by im-coder-lg** [comment]
Anyway, teardown results:
[TimerX.zip](https://github.com/sumeshir26/TimerX/files/7762237/TimerX.zip)
macOS seems to run Windows apps though(check the `TimerX.app/Contents/MacOS` directory)... Apple, what have you done here?

**Comment by im-coder-lg** [comment]
> If you run that, it successfully generates one for all platforms and creates a release successfully.

Can I test that? Just curious. Also, did you mark the release as a draft(in the YAML code)? If you didn't, I will, since we need to document the changes directly, not via the GitHub Actions. There's a new feature that allows you to automatically document all changes, so I will make sure it creates a draft release and try the auto release note feature. If it releases TimerX, I'd never ever use the auto notes feature.

**Comment by sumeshir26** [comment]
Yes, pease!

**Comment by im-coder-lg** [comment]
Well, failed in the macOS steps, fixing that in my branch. For the delay, I was checking my notifs and clearing some Vivaldi clutter.

**Comment by sumeshir26** [comment]
Reopening because not fixed

**Comment by not-nef** [comment]
okay, i thought you dropped macos support, my fault

**Comment by sumeshir26** [comment]
@Futura-Py/Reviewers

**Comment by im-coder-lg** [comment]
Sir, yes, sir! I just have to add some framework. Now my school's making me work into the actual regular timings, so devel would be really slow.

**Comment by not-nef** [comment]
Guys i just saw we have gh pages, so we wouldnt need to host our website on netlify!

**Comment by sumeshir26** [comment]
Yeah!

**Comment by not-nef** [comment]
mhm
